app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/SafeSettingsFragment.kt

```diff
@@ -26,11 +43,64 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.button.setOnClickListener {
-            viewModel.removeSafe()
+        binding.remove.setOnClickListener { showRemoveDialog() }
+        viewModel.state.observe(viewLifecycleOwner, Observer {
+            when (val viewAction = it.viewAction) {
+                is BaseStateViewModel.ViewAction.Loading -> loadDetails(viewAction.isLoading, it.safe, it.safeInfo, it.ensName)
```

Comment:
in other places Loading is used only to indicate the status of loading. Not to display data